draft-ietf-rats-eat.md

```diff
@@ -685,33 +714,81 @@ more security-oriented subsystems like a TEE or a Secure Element.
+identified by its specific label. It is a peer to other claims, but it
+is not called a claim because it is a container for a claim set rather
+than an individual claim. This submods part of a token allows what
+might be called recursion. It allows claim sets inside of claim sets
+inside of claims sets...
+
+### Two Types of Submodules
+
+Each entry in the submod map one of two types:
+
+* A non-token submodule that is a map or object directly containing
+  claims for the submodule.
+* A nested EAT that is a fully-formed, independently signed EAT token
+
+#### Non-token Submodules
 
-The security level of the submod is assumed to be at the same level as
-the main entity unless there is a security level claim in that
-submodule indicating otherwise. The security level of a submodule can
-never be higher (more secure) than the security level of the EAT it is
-a part of.
+Essentially this type of submodule, is just a sub-map or sub-object
+containing claims. It is recognized from the other type by being
+a data item of type map in CBOR or by being an object in JSON.
 
-### The submod_name Claim
+The contents are claims about the submodule of types defined 
+in this document or anywhere else claims types are defined.
 
-Each submodule should have a submod_name claim that is descriptive
-name. This name should be the CBOR txt type.
+#### Nested EATs
+
+This type of submodule is a fully formed EAT as described here. In
+this case the submodule has key material distinct from the containing
+EAT token that allows it to sign on its own.
+
+When an EAT is nested in another EAT as a submodule the nested EAT
+MUST use the CBOR CWT tag. This clearly distinguishes it from the
+non-token submodules.
+
+### No Inheritance
+
+The subordinate modules do not inherit anything from the containing
+token.  The subordinate modules must explicitly include all of their
+claims. This is the case even for claims like the nonce and age.
+
+This rule is in place for simplicity. It avoids complex inheritance
+rules that might vary from one type of claim to another. (TODO: fix
+the boot claim which does have inheritance as currently described).
+
+### Security Levels and Attachment
+
+The security level of the non-token subordinate modules should always
+be less than that of the containing modules in the case of non-token
+submodules. It makes no sense for a module of lesser security to be
+signing claims of a module of higher security. An example of this is a
+TEE signing claims made by the non-TEE parts (e.g. the high-level OS)
+of the device.
+
+The opposite may be true for the nested tokens. They usually have
+their own more secure key material. An example of this is an embedded
+secure element.
+
+The submods attachment claim describes how strong the connection is of
+a submodule to the containing module. For stronger attachment types
+the submod might choose to not include claims like the nonce for size,
+but this is not recommended.
```

Comment:
this seems to contradict line 755 which mandates the inclusion of all such claims.
Should there be an explicit statement (or claim value) for whether submod is expected to inherit or not rather than making it implicit on attachment type?

Reply:
Line 754 is explicit that there is no inheritance.

Agree that the wording here is fuzzy.

The nonce is intrinsically different from other claims as it does not describe the entity. It comes from the relying party to prevent replay. To replicate or not replicate the nonce could be considered to not be an inheritance question and we could have explicit text spelling all this out.